### PR TITLE
fixed search when entering some series it dosen't give a result

### PR DESCRIPTION
**Pull request by xOnlyFadi**

### CODE REVIEW

File: sources for users/sources-dev/readcomiconline/readcomiconline.js [hunk 1]
```diff
@@ -621,9 +621,35 @@ module.exports = class Readcomiconline extends Source  {
     searchMangaParse(page /*Int*/, response /*String -- latestUpdatesResponse -- currentPageHtml */, query, filters){
         var page = parseInt(page); //protect WKNodeBrowserify
         var searchMangaSelector = this.searchMangaSelector();
-        var json = [];
         console.log("readcomiconline searchMangaResponse -- ", response);
         var $ = cheerio.load(response);
+        var json = [];
+        var directManga = $('.barTitle',$('.rightBox')).first().text().trim()
```
Comment (chubimauk) [inline sources for users/sources-dev/readcomiconline/readcomiconline.js]: don't think this is safe

File: sources for users/sources-dev/readcomiconline/readcomiconline.js [hunk 2]
```diff
@@ -621,9 +621,35 @@ module.exports = class Readcomiconline extends Source  {
...
+        //checks if the comic was redirected and then makes one page mangapage
+        if (directManga === 'Cover'){
```
Comment (chubimauk) [inline sources for users/sources-dev/readcomiconline/readcomiconline.js]: why is this ===?

**Review by chubimauk — CHANGES_REQUESTED** [review]

File: sources for users/sources-dev/readcomiconline/readcomiconline.js (hunk 2 above)
Comment (xOnlyFadi) [reply sources for users/sources-dev/readcomiconline/readcomiconline.js]: when i use == it dosen't get the name and returns it as search result

File: sources for users/sources-dev/readcomiconline/readcomiconline.js (hunk 1 above)
Comment (xOnlyFadi) [reply sources for users/sources-dev/readcomiconline/readcomiconline.js]: i probably forgot fixing this

**Review by xOnlyFadi — COMMENTED** [review]
for the pages push i forgot to make it inside the if and for the === when i used == it did not work at all

File: sources for users/sources-dev/readcomiconline/readcomiconline.js (hunk 1 above)
Comment (chubimauk) [reply sources for users/sources-dev/readcomiconline/readcomiconline.js]: make it safe before I merge it

File: sources for users/sources-dev/readcomiconline/readcomiconline.js (hunk 2 above)
Comment (chubimauk) [reply sources for users/sources-dev/readcomiconline/readcomiconline.js]: no reason to use ===, change it. you can do `${directManga}` == 'Cover' as a last resort

File: sources for users/sources-dev/readcomiconline/readcomiconline.js (hunk 1 above)
Comment (xOnlyFadi) [reply sources for users/sources-dev/readcomiconline/readcomiconline.js]: When i go back home i will fix it

**Review by chubimauk — APPROVED** [review]

**Pull request merged**